### PR TITLE
Cleanup C++11 guards

### LINKED ISSUE #947: Clean up C++11 guards
All the `__cplusplus >= 201103L` needs to be cleaned up.

### PR DESCRIPTION
**Pull request by ye-luo**
Closes #947
with a bit extra cleaning of Xoen Phi/MIC accelerator.

### CODE REVIEW

File: src/config.h.cmake.in [hunk 1]
```diff
@@ -188,9 +188,9 @@
 /* Use SOA version of AA distance table */
 #cmakedefine ENABLE_SOA  @ENABLE_SOA@
 
-#if (__cplusplus >= 201103L)
+#if defined(__cplusplus)
   #if defined(__INTEL_COMPILER)
```
Comment (prckent) [inline src/config.h.cmake.in]: Why is special handling for the Intel compiler needed here? Are we hurting other compilers with inequivalent setting or working around an implementation gap?

Comment (ye-luo) [reply src/config.h.cmake.in]: Giving compiler information about alignment is not portable across compilers.
GNU/Clang use __builtin_assume_aligned
Intel uses __assume_aligned
This is the only difference here.

**Comment by prckent** [comment]
Are we missing proper clang alignment support? (Create an issue)

**Comment by ye-luo** [comment]
Clang sets the GNU flag and uses the GNU alignment setting.

**Pull request merged**